SU2_CFD/src/numerics/CNumerics.cpp

```diff
@@ -278,10 +279,20 @@ void CNumerics::GetInviscidIncProjJac(const su2double *val_density, const su2dou
     val_Proj_Jac_Tensor[2][2] = val_scale*((*val_density)*(proj_vel + val_normal[1]*val_velocity[1]));
     val_Proj_Jac_Tensor[2][3] = val_scale*((*val_dRhodT)*val_velocity[1]*proj_vel);
 
-    val_Proj_Jac_Tensor[3][0] = val_scale*((*val_cp)*(*val_temperature)*proj_vel/(*val_betainc2));
-    val_Proj_Jac_Tensor[3][1] = val_scale*((*val_cp)*(*val_temperature)*val_normal[0]*(*val_density));
-    val_Proj_Jac_Tensor[3][2] = val_scale*((*val_cp)*(*val_temperature)*val_normal[1]*(*val_density));
-    val_Proj_Jac_Tensor[3][3] = val_scale*((*val_cp)*((*val_temperature)*(*val_dRhodT) + (*val_density))*proj_vel);
+    if (energy_multicomponent) {
+      val_Proj_Jac_Tensor[3][0] = val_scale * ((*val_temperature) * proj_vel / (*val_betainc2));
+      val_Proj_Jac_Tensor[3][1] = val_scale * ((*val_temperature) * val_normal[0] * (*val_density));
+      val_Proj_Jac_Tensor[3][2] = val_scale * ((*val_temperature) * val_normal[1] * (*val_density));
+      val_Proj_Jac_Tensor[3][3] =
+          val_scale * ((*val_temperature) * (*val_dRhodT) + (*val_density)) * proj_vel;
```

Comment:
What is happening here? Is temperature being used for enthalpy?

Reply:
the energy can be $C_{p}T$ or enthalpy. The enthalpy is used not the temperature.

Reply:
then some renaming is in order and you need to come up with a better solution to switch between formulations that to have an if statement everytime Cp is used (for example making Cp=1, or encapsulating the computation of enthalpy in a function).